### ISSUE TITLE
[BUG] custom_pipeline in setup does not transform the dataset

### ISSUE THREAD

**Issue by mpaulonis** [pycaret/pycaret#1607]
Based on some discussion in the PyCaret Slack, I attempted to help a user employ custom_pipeline in setup to do some custom dataset preprocessing that they wanted to embed in the preprocessing pipeline such that it would be retained through model save/load operations.  They were unable to get custom_pipeline to complete their simple data transformation.  I looked at the code in tabular.py and it's not clear that custom_pipeline steps are ever provided a way to transform the data.  As a check I looked at the PyCaret test code for custom_pipeline, and it appears that even the test code does not work.  The test does not produce any errors, but it also does not produce the expected result.  Below is a subset of the custom_pipeline test code.

```
import pycaret.classification
import pycaret.datasets
import sklearn.pipeline
import sklearn.decomposition

data = pycaret.datasets.get_data("juice")
clf1 = pycaret.classification.setup(
        data,
        target="Purchase",
        log_experiment=True,
        silent=True,
        html=False,
        custom_pipeline=[("CUSTOM_PCA", sklearn.decomposition.PCA())],
        # pca=True,
        # pca_components=16,
        session_id=123,
        n_jobs=1,
)
test = pycaret.classification.get_config('X')
test.head()
```
The expected behavior is that the PCA() routine in the custom_pipeline should perform PCA on the juice dataset and the transformed X data from setup should be a set of principal component scores similar to the output that would occur if the commented arguments in setup were uncommented and the internal PCA preprocessing was performed.  The X data output from setup is just the default preprocessed juice dataset, with all the juice variables along with several dummy variables.

I realize that it could be a bit dangerous allowing custom transformers to be inserted before the built-in transformers, but the documentation indicates this is the purpose of this setup argument.

Also note that `get_config('prep_pipe')` shows the CUSTOM_PCA added at the end of the pipeline, whereas the documentation suggests that it would be added before any internal transformations.

**Versions**
2.3.3

**Comment by moezali1** [comment]
@Yard1 @tvdboom Something for you guys to look into.

**Comment by davcem** [comment]
I have discovered the same issue (in version 2.3.9) but with a different "use case": I want to compare a few feature selection methods and since they are not offered by pycaret I am using feature selectors (from scikit-learn) in custom_pipeline. After setup and compare models I get the error: 
`ValueError: X has 85 features per sample; expecting 5`
This is due to the situation, that transform of the fs in the pipeline is not called, therefor features "get not reduced" (or filtered).
Minimal working example: 
`from pycaret.classification import setup, compare_models
from sklearn.feature_selection import SelectKBest
from sklearn.feature_selection import f_classif
from pycaret.datasets import get_data

credit_data = get_data('credit')
custom_pipeline = ('feature_selector', SelectKBest(f_classif, k=5))
exp_clf101 = setup(data=credit_data,
                   target='default',
                   session_id=123,
                   silent=True,  # deactivate feature type confirmation
                   custom_pipeline=custom_pipeline
                   )
best_model = compare_models(include=['lr', 'dt'],
                            round=3,
                            cross_validation=False,
                            sort='MCC',
                            n_select=2,
                            errors="raise")`

This issue seems to affect "all types" of custom preprocessing.

**Comment by maxiuboldi** [comment]
> I have discovered the same issue (in version 2.3.9) but with a different "use case": I want to compare a few feature selection methods and since they are not offered by pycaret I am using feature selectors (from scikit-learn) in custom_pipeline. After setup and compare models I get the error:
> 
> `ValueError: X has 85 features per sample; expecting 5`
> 
> This is due to the situation, that transform of the fs in the pipeline is not called, therefor features "get not reduced" (or filtered). Minimal working example:
> 
> ```
> from pycaret.classification import setup, compare_models
> from sklearn.feature_selection import SelectKBest`
> from sklearn.feature_selection import f_classif`
> from pycaret.datasets import get_data
> 
> credit_data = get_data('credit')
> custom_pipeline = ('feature_selector', SelectKBest(f_classif, k=5))
> exp_clf101 = setup(data=credit_data,
>                    target='default',
>                    session_id=123,
>                    silent=True,  # deactivate feature type confirmation
>                    custom_pipeline=custom_pipeline
>                    )
> best_model = compare_models(include=['lr', 'dt'],
>                             round=3,
>                             cross_validation=False,
>                             sort='MCC',
>                             n_select=2,
>                             errors="raise")
> ```
> 
> This issue seems to affect "all types" of custom preprocessing.

I have the same issue...

**Comment by tvdboom** [comment]
solved in v3.0
